saleor/order/views.py

```diff
@@ -112,3 +112,26 @@ def cancel_payment(request, order):
             form.save()
         return redirect('order:payment', token=order.token)
     return HttpResponseForbidden()
+
+
+def create_password(request, token):
+    if request.user.is_authenticated():
+        return redirect('order:details', token=token)
+    order = get_object_or_404(Order, token=token)
+    email = order.user_email
+    form_data = request.POST.copy()
+    if form_data:
+        form_data.update({'email': email})
```

Comment:
No, it's here to ensure that newly created user has the same e-mail as the placed order. In theory, email could be changed in browser because we're using hidden input there (because we're reusing password form from registratin)